### PR TITLE
[Numpy] Numpy eye

### PR DESCRIPTION
**Pull request by stu1130**
## Description ##
Numpy eye

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/mxnet/ndarray/numpy/_op.py [hunk 1]
```diff
@@ -2363,3 +2363,32 @@ def var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):  # pylint:
     0.2025
     """
     return _npi.var(a, axis=axis, dtype=dtype, ddof=ddof, keepdims=keepdims, out=out)
+
+
+@set_module('mxnet.ndarray.numpy')
+def eye(N, M=None, k=0, dtype=_np.float32, **kwargs):
+    """
+    Return a 2-D array with ones on the diagonal and zeros elsewhere.
+    Parameters
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: One more blank line above this line.

File: python/mxnet/ndarray/numpy/_op.py [hunk 2]
```diff
@@ -2363,3 +2363,32 @@ def var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):  # pylint:
...
+        Number of rows in the output.
+    M : int, optional
+        Number of columns in the output. If None, defaults to N.
+    k : int, optional
+        Index of the diagonal: 0 (the default) refers to the main diagonal,
+        a positive value refers to an upper diagonal,
+        and a negative value to a lower diagonal.
+    dtype : data-type, optional
+        Data-type of the returned array.
+    Returns
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: One more blank line above this line.

File: python/mxnet/numpy/multiarray.py [hunk 3]
```diff
@@ -3808,3 +3808,28 @@ def var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=None):
     0.2025
     """
     return _npi.var(a, axis=axis, dtype=dtype, ddof=ddof, keepdims=keepdims, out=out)
+
+
+@set_module('mxnet.numpy')
+def eye(N, M=None, k=0, dtype=_np.float32, **kwargs):
+    """
+    Return a 2-D array with ones on the diagonal and zeros elsewhere.
+    Parameters
```
Comment (haojin2) [inline python/mxnet/numpy/multiarray.py]: One more blank line above this line.

File: python/mxnet/numpy/multiarray.py [hunk 4]
```diff
@@ -3808,3 +3808,28 @@ def var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=None):
...
+        Number of rows in the output.
+    M : int, optional
+        Number of columns in the output. If None, defaults to N.
+    k : int, optional
+        Index of the diagonal: 0 (the default) refers to the main diagonal,
+        a positive value refers to an upper diagonal,
+        and a negative value to a lower diagonal.
+    dtype : data-type, optional
+        Data-type of the returned array.
+    Returns
```
Comment (haojin2) [inline python/mxnet/numpy/multiarray.py]: One more blank line above this line.

File: python/mxnet/symbol/numpy/_symbol.py [hunk 5]
```diff
@@ -2678,4 +2678,33 @@ def var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):  # pylint:
     return _npi.var(a, axis=axis, dtype=dtype, ddof=ddof, keepdims=keepdims, out=out)
 
 
+@set_module('mxnet.symbol.numpy')
+def eye(N, M=None, k=0, dtype=_np.float32, **kwargs):
+    """
+    Return a 2-D array with ones on the diagonal and zeros elsewhere.
+    Parameters
```
Comment (haojin2) [inline python/mxnet/symbol/numpy/_symbol.py]: One more blank line above this line.

File: python/mxnet/symbol/numpy/_symbol.py [hunk 6]
```diff
@@ -2678,4 +2678,33 @@ def var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):  # pylint:
...
+        Number of rows in the output.
+    M : int, optional
+        Number of columns in the output. If None, defaults to N.
+    k : int, optional
+        Index of the diagonal: 0 (the default) refers to the main diagonal,
+        a positive value refers to an upper diagonal,
+        and a negative value to a lower diagonal.
+    dtype : data-type, optional
+        Data-type of the returned array.
+    Returns
```
Comment (haojin2) [inline python/mxnet/symbol/numpy/_symbol.py]: One more blank line above this line.

**Comment by stu1130** [comment]
@haojin2 done you can review again thanks

**Pull request merged**